rse-r-package.Rmd

```diff
@@ -81,134 +99,109 @@ An R package must contain the following files:
 usethis::use_code_of_conduct()
-usethis::use_citation()
 ```
 
-We then create a placeholder version of the function we want to write
-in a file called `R/frequency.R`
-either by using `File...New` in RStudio
-or by running `usethis::use_r('R/frequency.R')`:
+(Note that `use_mit_license` creates two files: `LICENSE` and `LICENSE.md`.
+The rules for R packages require the former,
+but GitHub expects the latter.)
+We then edit `README.md` to be:
 
-```{r, child="rse-r-package/etc/frequency-func.R"}
 ```
+# zipffreq
 
-and then:
+An example package for Merely Useful that checks Zipf's Law for classic English novels.
 
-1.  change the `Title`, `Author`, `Maintainer`, and `Description` in the `DESCRIPTION` file and
-2.  go to the `Build` tab in RStudio and run `Check` to see if our package meets CRAN's standards.
+## Installation
 
-```{r, child="rse-r-package/etc/DESCRIPTION-original.md"}
+TBD
+
+## Example
+
+FIXME: add an example.
 ```
 
-When we run `Check`,
-we get multiple lines of output with one warning and three notes
-in the summary at the bottom:
+and make a similar edit to `DESCRIPTION` so that it contains:
 
 ```
-FIXME: fill in with correct output
+Package: zipffreq
+Title: Checks Zipf's Law for classic English novels.
+Version: 0.0.0.9000
+Authors@R:
+    person(given = "Merely",
+           family = "Useful",
+           role = c("aut", "cre"),
+           email = "merely.useful@gmail.com")
+Description: >
+  Contains data and functions for checking Zipf's Law
+  for a set of classic English novels.
+License: MIT + file LICENSE
+Encoding: UTF-8
+LazyData: true
 ```
 
-The warning tells us that we haven't written any documentation for the `count_words` function.
-We also add lines to `.Rbuildignore` to tell RStudio to ignore `CITATION.md` and `CONDUCT.md`
-using `usethis::use_build_ignore`,
-after which `.Rbuildignore` looks like this:
+We can now go to the `Build` tab in RStudio and run `Check`
+to see if our empty package makes sense.
+When we do,
+the check warns us that there shouldn't be a period at the end of the package title.
+Once we fix that,
+we get a clean bill of health.
 
-```{r, child="rse-r-package/zipffreq/.Rbuildignore"}
-```
+> ### Leftovers
+>
+> Running `Check` creates another directory called `zipffreq.Rcheck`
+> and a file called `zipffreq_0.0.0.9000.tar.gz`.
+> These are created beside our project directory rather than in it
+> so as not to confuse version control---we don't want
+> the files we are building committed to our repository.
 
-We could change the name of `CITATION.md` to `inst/CITATION` instead,
-so that it will be copied into the root of the installation directory on users' machines,
-bit we decide to give priority to GitHub's convention that
-the citation description in the project's root directory.
+We can now create a placeholder for one of the functions we want to write
+in a file called `R/frequency.R`
+either by using `File...New` in RStudio
+or by running `usethis::use_r('R/frequency.R')`:
```

Comment:
`use_r()` will only create an R script inside the `R/` folder so prefixing `R/` is not necessary.